### PR TITLE
Reuse @com_google_protobuf//:protoc when generating scala from proto

### PR DESCRIPTION
**Pull request by simuons**
Remove `protoc-jar` which brings own versions of `protoc`

### CODE REVIEW

**Comment by simuons** [comment]
Resolves https://github.com/bazelbuild/rules_scala/issues/690

File: src/scala/scripts/PBGenerateRequest.scala [hunk 1]
```diff
@@ -2,7 +2,7 @@ package scripts
 
 import java.nio.file.{Files, Path, Paths}
 
-class PBGenerateRequest(val jarOutput: String, val scalaPBOutput: Path, val scalaPBArgs: List[String])
+class PBGenerateRequest(val jarOutput: String, val scalaPBOutput: Path, val scalaPBArgs: List[String], val protoc: String)
```
Comment (johnynek) [inline src/scala/scripts/PBGenerateRequest.scala]: Should protoc be a Path?

Comment (simuons) [reply src/scala/scripts/PBGenerateRequest.scala]: Changed to `Path`

**Comment by simuons** [comment]
@johnynek yes this is orthogonal to @ianoc-stripe's changes. These changes are not in his PR but he can take them and incorporate to his PR. Your call, I have no objections.

**Comment by simuons** [comment]
One side effect is that messages like

**Comment by johnynek** [comment]
cc @ianoc-stripe this seems like a good thing to copy in the aspect version.

**Pull request merged**